builder/googlecompute/config.go

```diff
@@ -115,6 +115,8 @@ type Config struct {
 	IAPConfig `mapstructure:",squash"`
 	// Skip creating the image. Useful for setting to `true` during a build test stage. Defaults to `false`.
 	SkipCreateImage bool `mapstructure:"skip_create_image" required:"false"`
+	// The architecture of the resulting image. Defaults to "x86_64".
+	ImageArchitecture string `mapstructure:"image_architecture" required:"false"`
```

Comment:
I'm starting to think about this, but since we're running the builder as an instance on GCE, could we maybe infer the image architecture from the instance type if it is unset, rather than defaulting to `x86_64`? This may not fit all cases to be fair though, so there might be an argument to be made in favour of defaulting to x86_64 since that's what we've supported until now.

Edit: been looking at the API docs, we could maybe lift stuff from the disks of the instance at first glance (the instance itself doesn't expose its architecture), but this happens later in the process so I'm not sure this is a clean way to infer that from. The `machineTypes` endpoint does not return the architecture of a machine as well (bit weird if you ask me). I presume we can default to x86_64 for now, this looks like the saner default